WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -0,0 +1,135 @@
+import Combine
+import UIKit
+import Yosemite
+import protocol Storage.StorageManagerType
+
+/// Coordinates navigation for store creation flow, with the assumption that the app is already authenticated with a WPCOM user.
+final class StoreCreationCoordinator: Coordinator {
+    /// Navigation source to store creation.
+    enum Source {
+        /// Initiated from the login prologue in logged-out state.
+        case prologue
+        /// Initiated from the store picker in logged-in state.
+        case storePicker
+    }
+
+    var navigationController: UINavigationController
+
+    @Published private var possibleSiteURLsFromStoreCreation: Set<String> = []
+    private var possibleSiteURLsFromStoreCreationSubscription: AnyCancellable?
+
+    private let source: Source
+    private let storePickerViewModel: StorePickerViewModel
+    private let switchStoreUseCase: SwitchStoreUseCaseProtocol
+
+    init(source: Source,
+         navigationController: UINavigationController,
+         storageManager: StorageManagerType = ServiceLocator.storageManager,
+         stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.source = source
+        self.navigationController = navigationController
+        self.storePickerViewModel = .init(configuration: .standard, //todo
```

Comment:
❓ Maybe we need the `.login` config here instead?

Reply:
the store picker is shown not only during the login process but also when the app is logged-in when switching stores or launching the app without a connected store. I think `standard` makes more sense here, so that non-Woo sites can be observed for checking if the new site is ready.

alternatively, we can update the `StorePickerViewModel`'s initializer to take in a parameter for whether to support non-Woo sites, since that seems to be the only place where the configuration is used. I just thought this feels a bit out of the scope of the PR.